doc/CONFIGURATION.md

```diff
@@ -175,6 +190,10 @@ Amazon S3 offers a [range of storage classes](https://aws.amazon.com/s3/storage-
 * `GLACIER` for [S3 Glacier Flexible Retrieval](https://aws.amazon.com/s3/storage-classes/glacier/)
 * `DEEP_ARCHIVE` for [S3 Glacier Deep Archive](https://aws.amazon.com/s3/storage-classes/glacier/)
 
+> [!IMPORTANT]
+> Do not set the storage class to `EXPRESS_ONEZONE` as it is a distinct storage class and cannot be set for general purpose
+buckets. If you want to use S3 Express One Zone storage class, just specify a directory bucket name when mounting.
```

Comment:
Is this something that could be enforced by code instead of relying on this documentation?

Reply:
This value is still valid for directory buckets, but it's the same as not setting storage class at all. So, maybe we can just don't allow people to set `--storage-class` to `EXPRESS_ONEZONE`.

Reply:
Users also cannot set storage class other than `EXPRESS_ONEZONE` for directory buckets. I think the right time to do the validation is after we have resolved the endpoint and know whether this is a directory bucket or not.

Reply:
FWIW, I'd caution against setting limits like this that would normally be enforced service-side in S3 but could feasibly be relaxed in the future. In adding this restriction to the client, users are forced to upgrade version to adopt new features (like a new storage class for directory buckets).

Maybe forcing updates has its upsides from maintainers' point of view, but I'd lean against it for now. 😄

Reply:
What's important is that the user gets clear feedback into what went wrong. If the wrong storage class is set, is the error clear in the Mountpoint logs? Of course we're always limited by the error responses we can send via filesystem APIs.